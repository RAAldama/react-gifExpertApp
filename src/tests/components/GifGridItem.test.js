import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Test on <GifGridItem />', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/something.jpg'
    const wrapper = shallow( <GifGridItem title={title} url={url} /> );
    
    test('should show <GifGridItem /> correctly', () => {   

        expect(wrapper).toMatchSnapshot();

    })
    

    test('should have a paragraph with text in the title', () => {
        
        const paragraph = wrapper.find('p');

        expect( paragraph.text().trim() ).toBe( title );

    })

    test('should have the image = to the url and alt of the props', () => {
        
        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    })

    test('should have class animated__fadeIn', () => {
        
        const div = wrapper.find('div');

        expect( div.prop('className').includes('animate__fadeIn') ).toBe(true);

    })
    
})
