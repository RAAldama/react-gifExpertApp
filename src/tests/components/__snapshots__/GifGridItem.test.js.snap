// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifGridItem /> should show <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/something.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
